packages/@react-aria/utils/src/scrollIntoView.ts

```diff
@@ -63,3 +71,33 @@ function relativeOffset(ancestor: HTMLElement, child: HTMLElement, axis: 'left'|
   }
   return sum;
 }
+
+export function scrollIntoViewport(targetElement: Element, containingElement?: Element) {
+  // If scrolling is not currently prevented then we aren’t in a overlay nor is a overlay open, just use element.scrollIntoView to bring the element into view
+  if (!isScrollPrevented) {
+    let {left: originalLeft, top: originalTop} = targetElement.getBoundingClientRect();
+
+    // TODO: For some reason in the useSelectableList stories, originalTop and newTop differ by 1px, causing us to unecessarily
+    // scroll the containing element even though it may already be fully in view
+    // This is because the scrollable parent (ul) has a border of 1 that partially obscures the focused item even after scrollIntoView is called...
+    // Perhaps modify scrollIntoView calculations to take into account the borders, bleh...
+
+    // use scrollIntoView({block: 'nearest'}) instead of .focus to check if the element is fully in view or not since .focus()
+    // won't cause a scroll if the element is already focused and doesn't behave consistently when an element is partially out of view horizontally vs vertically
+    targetElement?.scrollIntoView?.({block: 'nearest'});
```

Comment:
Cool that this works. I had assumed that `scrollIntoView` would always align the element, but looks like "nearest" doesn't scroll if it is already in view?

Reply:
Right, from my testing `scrollIntoView({block: 'nearest'})` seems to behave like `scrollIntoViewIfNeeded`